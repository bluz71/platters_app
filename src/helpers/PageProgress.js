import NProgress from 'nprogress'

class PageProgress {
  constructor () {
    this.loaded = false
  }

  start () {
    if (!this.loaded) {
      NProgress.start()
    }
  }

  done () {
    if (!this.loaded) {
      this.loaded = true
      NProgress.done()
    }

    return true
  }
}

export default PageProgress
