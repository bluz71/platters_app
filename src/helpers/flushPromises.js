// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

export default flushPromises;
