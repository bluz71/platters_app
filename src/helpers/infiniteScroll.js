// Code snippet based on:
//   http://blog.sodhanalibrary.com/2016/08/detect-when-user-scrolls-to-bottom-of.html
//
const infiniteScoll = (pagination, onPageEnd) => {
  const moreToScroll = pagination && pagination.next_page;
  if (!moreToScroll) {
    return;
  }

  const windowHeight = window.innerHeight;
  const body = document.body;
  const html = document.documentElement;
  const docHeight =
    Math.max(
      body.scrollHeight,
      body.offsetHeight,
      html.clientHeight,
      html.scrollHeight,
      html.offsetHeight
    ) - 20;
  const windowBottom = windowHeight + window.pageYOffset;

  if (windowBottom >= docHeight) {
    onPageEnd();
  }
};

export default infiniteScoll;
