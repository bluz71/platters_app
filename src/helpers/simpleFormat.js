// Code snippet provided by:
//   https://makandracards.com/makandra/1395-simple_format-helper-for-javascript
//
const simpleFormat = str => {
  str = str.replace(/\r\n?/, '\n')
  str = str.trim()

  if (str.length > 0) {
    str = str.replace(/\n\n+/g, '</p><p>')
    str = str.replace(/\n/g, '<br />')
    str = `<p>${str}</p>`
  }

  return str
}

export default simpleFormat
