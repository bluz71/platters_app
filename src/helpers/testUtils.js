import { appAuth } from '../lib/appAuth';

// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856
const flushPromises = () => new Promise(resolve => setImmediate(resolve));

const logInUser = () => (
  appAuth.logIn('eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjo1MywibmFtZSI6ImRlbm5pc2I1NSI\
sImFkbWluIjpmYWxzZSwiZXhwIjoxNTM3ODQ4NDA0LCJpc3MiOiJwbGF0dGVycyIsImF1ZCI6InBs\
YXR0ZXJzX2FwcCJ9.ZUV69wWVXfiiMtQ2ZFPvgqvhZWwQ4WC1bI15Q9DpFCU')
);

const logOutUser = () => (
  appAuth.logOut()
);

export { flushPromises, logInUser, logOutUser };
