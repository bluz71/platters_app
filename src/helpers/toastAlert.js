import { toast } from 'react-toastify';

const toastAlert = (message) => {
  toast.error(message, { className: 'ToastAlert' });
};

export default toastAlert;
