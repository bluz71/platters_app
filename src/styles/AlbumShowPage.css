.AlbumShow a {
  color: #515151;
  text-decoration: none;
}

.AlbumShow a:hover {
  color: #3c9a5f;
}

.AlbumShow .large-cover img {
  height: 315px;
  width: 315px;
  margin-top: 1em;
  margin-bottom: 1em;
  border: 1px solid #777;
  border-radius: 4px;
}

.AlbumShow .album-data h2 {
  display: inline;
  font-size: 1.8em;
}

.AlbumShow .album-data .icon {
  font-size: 0.8em;
  float: right;
  color: #515151;
  margin: 1em 0.5em;
}

.AlbumShow .album-data .icon .filter {
  color: #515151;
  cursor: pointer;
}

.AlbumShow .album-data .icon .filter:hover {
  color: #3c9a5f;
}

.AlbumShow .album-data table {
  margin-bottom: 0;
}

.AlbumShow .album-data tbody .visible {
  display: table-row;
}

.AlbumShow .album-data tbody .invisible {
  display: none;
}

.AlbumShow .album-data .tracks-gradient {
  position: absolute;
  bottom: 3em;
  left: 0;
  width: 100%;
  height: 10em;
  background-image: linear-gradient(to bottom, rgba(255, 255, 255, 0), #fff);
}

.AlbumShow .album-data .tracks-less {
  margin-top: 0.5em;
}

.AlbumShow .album-data .tracks-less:before {
  font-family: 'FontAwesome';
  font-size: 0.9em;
  padding-right: 1em;
  content: '\f102';
}

.AlbumShow .album-data .tracks-more {
  margin-top: 0.5em;
}

.AlbumShow .album-data .tracks-more:before {
  font-family: 'FontAwesome';
  font-size: 0.9em;
  padding-right: 1em;
  content: '\f103';
}

.AlbumShow .comments .page-header h1 {
  font-size: 26px;
}

.AlbumShow .comments .page-header h1 small {
  font-size: 65%;
}

.AlbumShow .comments h4 {
  margin-bottom: 1em;
}
