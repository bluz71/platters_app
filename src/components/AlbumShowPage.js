import React, { Component } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { Row, Col, PageHeader, Table, Button } from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import axios from 'axios';
import numeral from 'numeral';
import pluralize from 'pluralize';
import { API_HOST } from '../config';
import pageProgress from '../helpers/pageProgress';
import infiniteScroll from '../helpers/infiniteScroll';
import toastAlert from '../helpers/toastAlert';
import CommentsList from './CommentsList';
import '../styles/AlbumShowPage.css';

class AlbumShowPage extends Component {
  constructor(props) {
    super(props);

    this.artistSlug       = props.match.params.artist_id;
    this.albumSlug        = props.match.params.album_id;
    this.albumEndPoint    = `${API_HOST}/${this.artistSlug}/${this.albumSlug}.json`;
    this.showingAllTracks = false;
    this.scrollLessTracks = false;
    this.loaded           = false;
    this.commentsEndPoint = `${API_HOST}/${this.artistSlug}/${this.albumSlug}/comments.json`;
    this.waiting          = false; // For comments when infinite-scrolling.
    this.scrollToComments =
      (props.location.state && props.location.state.scrollToComments) || false;
    this.pageProgress     = new pageProgress();

    this.state = {
      album: {},
      tracks: [],
      comments: [],
      commentsPagination: {},
      notFound: false,
      error: null
    };

    // Bind 'this' for callback functions.
    this.handleTracksVisibility = this.handleTracksVisibility.bind(this);
    this.handleScroll           = this.handleScroll.bind(this);
    this.handlePageEnd          = this.handlePageEnd.bind(this);
  }

  componentDidMount() {
    window.onscroll = this.handleScroll;
    this.getAlbum(true);
  }

  componentWillUnmount() {
    window.onscroll = null;
  }

  componentDidUpdate() {
    if (this.scrollLessTracks) {
      this.scrollLessTracks = false;
      this.tracksAnchor.scrollIntoView(false);
    }
  }

  handleScroll() {
    infiniteScroll(this.state.commentsPagination, this.handlePageEnd);
  }

  handlePageEnd() {
    // Page has been scrolled to the end, hence retrieve the next set of
    // comments.

    // Disable scroll handling whilst records are being retrieved.
    window.onscroll = null;
    const commentsPageEndPoint
      = `${this.commentsEndPoint}?page=${this.state.commentsPagination.next_page}`;
    this.waiting = true;
    this.forceUpdate(); // Render spinner
    this.getComments(commentsPageEndPoint);
  }

  handleYear(year) {
    const params = { year };
    this.props.history.push('/albums', params);
  }

  handleGenre(genre) {
    const params = { genre };
    this.props.history.push('/albums', params);
  }

  handleTracksVisibility() {
    this.showingAllTracks = !this.showingAllTracks;
    if (!this.showingAllTracks) {
      this.scrollLessTracks = true;
    }
    this.forceUpdate();
  }

  getAlbum(scrollToTop = false) {
    axios.get(this.albumEndPoint)
      .then(response => {
        this.loaded = this.pageProgress.done();
        document.title = response.data.album.title;
        this.setState({
          album: response.data.album,
          tracks: response.data.tracks,
          comments: response.data.comments,
          commentsPagination: response.data.comments_pagination
        });
        if (this.scrollToComments) {
          this.scrollToComments = false;
          this.commentsAnchor.scrollIntoView();
        }
        else if (scrollToTop) {
          window.scrollTo(0, 0);
        }
      }).catch(error => {
        this.pageProgress.done();
        if (error.response && error.response.status === 404) {
          this.setState({ notFound: true });
        }
        else {
          this.setState({ error: error });
        }
      });
  }

  getComments(commentsEndPoint) {
    axios.get(commentsEndPoint)
      .then(response => {
        this.waiting = false;
        if (!window.onscroll) {
          // Re-enable scroll handling now that the records have been
          // retrieved.
          window.onscroll = this.handleScroll;
        }
        this.setState({
          comments: [...this.state.comments, ...response.data.comments],
          commentsPagination: response.data.pagination,
        });
      }).catch(error => {
        this.setState({ error: error });
      });
  }

  albumRetrieved() {
    if (!this.loaded || this.state.notFound || this.state.error) {
      return false;
    }
    else {
      return true;
    }
  }

  renderHeader() {
    if (this.state.notFound) {
      toastAlert(`The album ${this.albumSlug} does not exist`);
      return <Redirect to="/albums" />;
    }
    if (this.state.error) {
      toastAlert('Connection failure, please retry again soon');
      return <Redirect to="/" />;
    }

    const albumTitle = this.state.album.title;
    const artistName = this.state.album.artist_name;

    return (
      <Col md={10} mdOffset={1}>
        <PageHeader>
          {albumTitle} <small>by <Link 
            to={`/artist/${this.artistSlug}`}>{artistName}</Link></small>
        </PageHeader>
      </Col>
    );
  }

  renderCover() {
    const { album } = this.state;

    return (
      <Col md={4} mdOffset={1} className="large-cover">
        <img 
          className="img-responsive center-block" 
          src={album.cover_url} 
          alt={album.title}
        />
      </Col>
    );
  }

  trackVisibilityVal(track) {
    if (track.number <= 20 || this.showingAllTracks) {
      return 'visible';
    }
    else {
      return 'invisible';
    }
  }

  renderTracks(tracks) {
    return (
      tracks.map(track => {
        return (
          <tr key={track.id} className={this.trackVisibilityVal(track)}>
            <td>{track.number}.</td>
            <td>{track.title}</td>
            <td>{track.duration}</td>
          </tr>
        );
      })
    );
  }

  renderTracksVisibility() {
    const { tracks } = this.state;

    if (tracks.length <= 20) {
      return;
    }

    const btnText  = this.showingAllTracks ? 'Show less tracks' : 'Show all tracks';

    return (
      <div ref={tracksAnchor => this.tracksAnchor = tracksAnchor}>
        {!this.showingAllTracks && <div className="tracks-gradient" />}
        <Button
          bsSize="small"
          className={this.showingAllTracks ? 'tracks-less' : 'tracks-more'}
          onClick={this.handleTracksVisibility}
        >
          {btnText}
        </Button>
      </div>
    );
  }

  renderAlbum() {
    const { album, tracks } = this.state;

    return (
      <Col md={6} className="album-data">
        <h2>
          {album.tracks_count} Tracks <small>(Time {album.total_duration})</small>
        </h2>
        <span className="icon">
          <a onClick={() => this.handleYear(album.year)}>
            <FontAwesome name="calendar" /> {album.year}
          </a>
          <a onClick={() => this.handleGenre(album.genre)}>
            <FontAwesome name="tag" className="spacer-left-xs" /> {album.genre}
          </a>
        </span>
        <Table striped>
          <thead>
            <tr>
              <th>No.</th>
              <th>Title</th>
              <th>Duration</th>
            </tr>
          </thead>
          <tbody>
            {this.renderTracks(tracks)}
          </tbody>
        </Table>
        {this.renderTracksVisibility()}
      </Col>
    );
  }

  renderCommentsList(count) {
    if (count === 0) {
      return (
        <h4>No comments have been posted for this album</h4>
      );
    }

    return (
      <CommentsList comments={this.state.comments} shortHeader />
    );
  }

  renderSpinner() {
    if (this.waiting) {
      return (
        <div className="WaitingSpinner">
          <FontAwesome name="spinner" spin pulse />
        </div>
      );
    }
  }

  renderComments() {
    const count = this.state.commentsPagination.total_count;
    const commentsCount = numeral(count).format('0,0');

    return (
      <Col md={10} mdOffset={1} className="comments">
        <div ref={commentsAnchor => this.commentsAnchor = commentsAnchor}>
          <PageHeader>
            Comments {this.albumRetrieved()
                && <small>({commentsCount} {pluralize('Comment', count)})</small>}
          </PageHeader>
        </div>
        {this.renderCommentsList(count)}
        {this.renderSpinner()}
      </Col>
    );
  }

  render() {
    this.pageProgress.start();

    return (
      <Row>
        <div className="AlbumShow">
          {this.renderHeader()}
          {this.renderCover()}
          {this.renderAlbum()}
          {this.renderComments()}
        </div>
      </Row>
    );
  }
}

export default AlbumShowPage;
