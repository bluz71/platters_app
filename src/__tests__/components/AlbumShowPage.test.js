import React from 'react';
import { shallow } from 'enzyme';
import { flushPromises } from '../../helpers/testUtils';
import AlbumShowPage from '../../components/AlbumShowPage';

describe('<AlbumShowPage />', () => {
  it('renders album show page', async () => {
    const match = { params: { artist_id: 'abc', album_id: 'def' } };
    const wrapper = shallow(<AlbumShowPage match={match} location />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders all tracks when show all is clicked', async () => {
    const match = { params: { artist_id: 'abc', album_id: 'def' } };
    const wrapper = shallow(<AlbumShowPage match={match} location />);
    await flushPromises();
    wrapper.instance().handleTracksVisibility();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders less tracks when show less is clicked', async () => {
    const match = { params: { artist_id: 'abc', album_id: 'def' } };
    const wrapper = shallow(<AlbumShowPage match={match} location />);
    await flushPromises();
    wrapper.instance().handleTracksVisibility();
    wrapper.instance().handleTracksVisibility();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders all album comments when scrolled', async () => {
    const match = { params: { artist_id: 'abc', album_id: 'def' } };
    const wrapper = shallow(<AlbumShowPage match={match} location />);
    await flushPromises();
    wrapper.instance().handlePageEnd();
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('handles invalid album', async () => {
    const match = { params: { artist_id: 'abc', album_id: 'null' } };
    const wrapper = shallow(<AlbumShowPage match={match} location />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
