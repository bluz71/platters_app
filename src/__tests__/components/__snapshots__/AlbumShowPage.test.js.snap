// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumShowPage /> renders album show page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <div
    className="AlbumShow"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
      mdOffset={1}
    >
      <PageHeader
        bsClass="page-header"
      >
        DEF
         
        <small>
          by 
          <Link
            replace={false}
            to="/artist/abc"
          >
            ABC
          </Link>
        </small>
      </PageHeader>
    </Col>
    <Col
      bsClass="col"
      className="large-cover"
      componentClass="div"
      md={4}
      mdOffset={1}
    >
      <img
        alt="DEF"
        className="img-responsive center-block"
        src="http://localhost:3000/uploads/album/cover/1/ABC--DEF.jpg"
      />
    </Col>
    <Col
      bsClass="col"
      className="album-data"
      componentClass="div"
      md={6}
    >
      <h2>
        10
         Tracks 
        <small>
          (Time 
          45:55
          )
        </small>
      </h2>
      <span
        className="icon"
      >
        <a
          onClick={[Function]}
        >
          <FontAwesome
            name="calendar"
          />
           
          2002
        </a>
        <a
          onClick={[Function]}
        >
          <FontAwesome
            className="spacer-left-xs"
            name="tag"
          />
           
          Rock
        </a>
      </span>
      <Table
        bordered={false}
        bsClass="table"
        condensed={false}
        hover={false}
        responsive={false}
        striped={true}
      >
        <thead>
          <tr>
            <th>
              No.
            </th>
            <th>
              Title
            </th>
            <th>
              Duration
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="visible"
            key="1"
          >
            <td>
              1
              .
            </td>
            <td>
              First
            </td>
            <td>
              4:22
            </td>
          </tr>
          <tr
            className="visible"
            key="2"
          >
            <td>
              2
              .
            </td>
            <td>
              Second
            </td>
            <td>
              4:34
            </td>
          </tr>
          <tr
            className="visible"
            key="3"
          >
            <td>
              3
              .
            </td>
            <td>
              Third
            </td>
            <td>
              3:53
            </td>
          </tr>
          <tr
            className="visible"
            key="4"
          >
            <td>
              4
              .
            </td>
            <td>
              Fourth
            </td>
            <td>
              3:52
            </td>
          </tr>
          <tr
            className="visible"
            key="5"
          >
            <td>
              5
              .
            </td>
            <td>
              Fifth
            </td>
            <td>
              6:22
            </td>
          </tr>
          <tr
            className="visible"
            key="6"
          >
            <td>
              6
              .
            </td>
            <td>
              Sixth
            </td>
            <td>
              4:06
            </td>
          </tr>
          <tr
            className="visible"
            key="7"
          >
            <td>
              7
              .
            </td>
            <td>
              Seventh
            </td>
            <td>
              5:39
            </td>
          </tr>
          <tr
            className="visible"
            key="8"
          >
            <td>
              8
              .
            </td>
            <td>
              Eighth
            </td>
            <td>
              4:44
            </td>
          </tr>
          <tr
            className="visible"
            key="9"
          >
            <td>
              9
              .
            </td>
            <td>
              Ninth
            </td>
            <td>
              4:19
            </td>
          </tr>
          <tr
            className="visible"
            key="10"
          >
            <td>
              10
              .
            </td>
            <td>
              Tenth
            </td>
            <td>
              4:04
            </td>
          </tr>
        </tbody>
      </Table>
    </Col>
    <Col
      bsClass="col"
      className="album-comments"
      componentClass="div"
      id="comments"
      md={10}
      mdOffset={1}
    >
      <div>
        <PageHeader
          bsClass="page-header"
        >
          Comments 
          <small>
            (
            36
             
            Comments
            )
          </small>
        </PageHeader>
      </div>
      <CommentsList
        comments={
          Array [
            Object {
              "body": "First comment",
              "created_at": "<time datetime=\\"2018-06-28T08:56:53Z\\" data-local=\\"time-ago\\">June 28, 2018  8:56am</time>",
              "gravatar_url": "https://gravatar.com/avatar/e16475b708a5b231b5fdf71ab3577761?s=80&r=pg&d=identicon",
              "id": 29995,
              "user_name": "fermin_gottlieb83",
              "user_slug": "fermin_gottlieb83",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-06-23T09:25:36Z\\" data-local=\\"time-ago\\">June 23, 2018  9:25am</time>",
              "gravatar_url": "https://gravatar.com/avatar/61ac49478ea8c40129c5abc30153880d?s=80&r=pg&d=identicon",
              "id": 29991,
              "user_name": "anahi26",
              "user_slug": "anahi26",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-05-28T08:27:27Z\\" data-local=\\"time-ago\\">May 28, 2018  8:27am</time>",
              "gravatar_url": "https://gravatar.com/avatar/06ecce7fb50e191bcdfe06e324882fe0?s=80&r=pg&d=identicon",
              "id": 29984,
              "user_name": "rosalind_mante11",
              "user_slug": "rosalind_mante11",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-05-10T12:10:03Z\\" data-local=\\"time-ago\\">May 10, 2018 12:10pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/22d70f1821be0b4669a186e4ab18ff51?s=80&r=pg&d=identicon",
              "id": 30013,
              "user_name": "devin16",
              "user_slug": "devin16",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-04-30T02:03:15Z\\" data-local=\\"time-ago\\">April 30, 2018  2:03am</time>",
              "gravatar_url": "https://gravatar.com/avatar/9a8ad7cda90d364b0024667ecdf773d6?s=80&r=pg&d=identicon",
              "id": 29980,
              "user_name": "lexi_jones94",
              "user_slug": "lexi_jones94",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-04-26T08:45:34Z\\" data-local=\\"time-ago\\">April 26, 2018  8:45am</time>",
              "gravatar_url": "https://gravatar.com/avatar/22d70f1821be0b4669a186e4ab18ff51?s=80&r=pg&d=identicon",
              "id": 29985,
              "user_name": "devin16",
              "user_slug": "devin16",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-04-09T20:05:33Z\\" data-local=\\"time-ago\\">April  9, 2018  8:05pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/d2ab58d215ee0d4bb9121f757e483944?s=80&r=pg&d=identicon",
              "id": 30012,
              "user_name": "ernesto31",
              "user_slug": "ernesto31",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-03-27T00:27:27Z\\" data-local=\\"time-ago\\">March 27, 2018 12:27am</time>",
              "gravatar_url": "https://gravatar.com/avatar/503710afd68777eeb329e1be844c140e?s=80&r=pg&d=identicon",
              "id": 30000,
              "user_name": "rollin50",
              "user_slug": "rollin50",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-03-15T05:38:12Z\\" data-local=\\"time-ago\\">March 15, 2018  5:38am</time>",
              "gravatar_url": "https://gravatar.com/avatar/3fff9f36bf92b24cd899eeb6cea05ff6?s=80&r=pg&d=identicon",
              "id": 29993,
              "user_name": "dawn_parker79",
              "user_slug": "dawn_parker79",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-03-14T19:37:15Z\\" data-local=\\"time-ago\\">March 14, 2018  7:37pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/48c30a3c1f69a8ebba162afbcaab7c67?s=80&r=pg&d=identicon",
              "id": 30006,
              "user_name": "kennith_rolfson66",
              "user_slug": "kennith_rolfson66",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-03-13T14:42:54Z\\" data-local=\\"time-ago\\">March 13, 2018  2:42pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/ba14e202ce8ffea35bcf9971bf0d6e85?s=80&r=pg&d=identicon",
              "id": 29992,
              "user_name": "elmore16",
              "user_slug": "elmore16",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-02-28T21:42:35Z\\" data-local=\\"time-ago\\">February 28, 2018  9:42pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/62279fdb4a74d49ae14c5bdbfbbef55f?s=80&r=pg&d=identicon",
              "id": 30007,
              "user_name": "bettie54",
              "user_slug": "bettie54",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-02-24T14:00:09Z\\" data-local=\\"time-ago\\">February 24, 2018  2:00pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/d7de8c9ee403ef8d8403429ba3c4ba11?s=80&r=pg&d=identicon",
              "id": 30001,
              "user_name": "jennifer28",
              "user_slug": "jennifer28",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-02-19T19:11:12Z\\" data-local=\\"time-ago\\">February 19, 2018  7:11pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/f5a520fe87dae7dbf7ce8869eda7af0f?s=80&r=pg&d=identicon",
              "id": 29994,
              "user_name": "ally_shields57",
              "user_slug": "ally_shields57",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-02-02T21:59:06Z\\" data-local=\\"time-ago\\">February  2, 2018  9:59pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/55629d25652543219eabe63afb23be16?s=80&r=pg&d=identicon",
              "id": 29978,
              "user_name": "orlando36",
              "user_slug": "orlando36",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-01-23T17:46:36Z\\" data-local=\\"time-ago\\">January 23, 2018  5:46pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/d7de8c9ee403ef8d8403429ba3c4ba11?s=80&r=pg&d=identicon",
              "id": 30004,
              "user_name": "jennifer28",
              "user_slug": "jennifer28",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-01-04T06:53:20Z\\" data-local=\\"time-ago\\">January  4, 2018  6:53am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7623bc3892c4bf9c2571447e23ec565f?s=80&r=pg&d=identicon",
              "id": 30008,
              "user_name": "leonel_kertzmann31",
              "user_slug": "leonel_kertzmann31",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2018-01-01T19:59:24Z\\" data-local=\\"time-ago\\">January  1, 2018  7:59pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/503710afd68777eeb329e1be844c140e?s=80&r=pg&d=identicon",
              "id": 29981,
              "user_name": "rollin50",
              "user_slug": "rollin50",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-12-27T19:28:39Z\\" data-local=\\"time-ago\\">December 27, 2017  7:28pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/0dd9e5ca83217d75bc53ae4fe0d76e06?s=80&r=pg&d=identicon",
              "id": 30010,
              "user_name": "arvel_ko76",
              "user_slug": "arvel_ko76",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-12-03T00:09:56Z\\" data-local=\\"time-ago\\">December  3, 2017 12:09am</time>",
              "gravatar_url": "https://gravatar.com/avatar/d7de8c9ee403ef8d8403429ba3c4ba11?s=80&r=pg&d=identicon",
              "id": 29989,
              "user_name": "jennifer28",
              "user_slug": "jennifer28",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-12-02T10:28:32Z\\" data-local=\\"time-ago\\">December  2, 2017 10:28am</time>",
              "gravatar_url": "https://gravatar.com/avatar/3ab954c11fb693fc8f966e9b5ab9f205?s=80&r=pg&d=identicon",
              "id": 29979,
              "user_name": "iva83",
              "user_slug": "iva83",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-11-23T07:57:12Z\\" data-local=\\"time-ago\\">November 23, 2017  7:57am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7f9f0846a96f762c757db392bd6a1588?s=80&r=pg&d=identicon",
              "id": 29988,
              "user_name": "winfield1",
              "user_slug": "winfield1",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-11-17T18:19:48Z\\" data-local=\\"time-ago\\">November 17, 2017  6:19pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/975cff4dfaf3813a2c8803e751353808?s=80&r=pg&d=identicon",
              "id": 30002,
              "user_name": "claria_klein19",
              "user_slug": "claria_klein19",
            },
            Object {
              "body": "Comment",
              "created_at": "<time datetime=\\"2017-11-04T04:31:56Z\\" data-local=\\"time-ago\\">November  4, 2017  4:31am</time>",
              "gravatar_url": "https://gravatar.com/avatar/7623bc3892c4bf9c2571447e23ec565f?s=80&r=pg&d=identicon",
              "id": 29998,
              "user_name": "leonel_kertzmann31",
              "user_slug": "leonel_kertzmann31",
            },
            Object {
              "body": "Last Comment",
              "created_at": "<time datetime=\\"2017-11-01T12:26:44Z\\" data-local=\\"time-ago\\">November  1, 2017 12:26pm</time>",
              "gravatar_url": "https://gravatar.com/avatar/188deba48d739c0287b061713d2c16be?s=80&r=pg&d=identicon",
              "id": 30003,
              "user_name": "simeon59",
              "user_slug": "simeon59",
            },
          ]
        }
        shortHeader={true}
      />
    </Col>
  </div>
</Row>
`;
